File: src/my_py_pkg/my_py_pkg/services/add_two_ints_client.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts

class AddTwoIntsClientNode(Node):
    def __init__(self):
        super().__init__('add_two_ints_client')
        self.client = self.create_client(AddTwoInts, 'add_two_ints')
        # wait for server (can also be added as the first code in send_request())
        while not self.client.wait_for_service(timeout_sec=1.0):
            self.get_logger().info('Waiting for add_two_ints service...')

    def send_request(self, a, b):
        request = AddTwoInts.Request()
        request.a = a
        request.b = b
        future = self.client.call_async(request)
        future.add_done_callback(self.callback_add_two_ints_response)
        

    def callback_add_two_ints_response(self, future):
        repsonse = future.result()
        self.get_logger().info(f"got response: {repsonse.sum}")

def main(args=None):
    rclpy.init(args=args)
 
    a = 2
    b = 3
    node = AddTwoIntsClientNode()
    node.send_request(a, b)

    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: src/my_cpp_pkg/src/services/add_two_ints_client.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"

class AddTwointsClientNode : public rclcpp::Node // MODIFY NAME
{
public:
    AddTwointsClientNode() : Node("add_two_ints_client") // MODIFY NAME
    {
        client_ = this->create_client<example_interfaces::srv::AddTwoInts>("add_two_ints");
    }

    void callAddTwoInts(int a, int b){
        while (!client_->wait_for_service(std::chrono::seconds(1))){
            RCLCPP_WARN(this->get_logger(), "waiting for server...");
        }
        auto request = std::make_shared<example_interfaces::srv::AddTwoInts::Request>();
        request->a = a;
        request->b = b;

        auto callback = std::bind(&AddTwointsClientNode::callback_addTwoInts, this, std::placeholders::_1);
        client_->async_send_request(request, callback);
    }

private:
    void callback_addTwoInts(rclcpp::Client<example_interfaces::srv::AddTwoInts>::SharedFuture future){
        auto response = future.get();
        RCLCPP_INFO(this->get_logger(), "Sum: %d", (int)response->sum);
    }

    std::shared_ptr<rclcpp::Client<example_interfaces::srv::AddTwoInts>> client_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<AddTwointsClientNode>(); // MODIFY NAME
    node->callAddTwoInts(6, 9);
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}
